rpcs3/rpcs3qt/game_compatibility.h

```diff
@@ -8,22 +8,116 @@
 class downloader;
 class gui_settings;
 
-struct compat_status
+namespace compat
 {
-	int index;
-	QString date;
-	QString color;
-	QString text;
-	QString tooltip;
-	QString latest_version;
-};
+	/** Represents the "title" json object */
+	struct pkg_title
+	{
+		std::string type; // TITLE or TITLE_08 etc. (system languages)
+		std::string title; // The Last of Arse
+	};
+
+	/** Represents the "changelog" json object */
+	struct pkg_changelog
+	{
+		std::string type; // paramhip or paramhip_08 etc. (system languages)
+		std::string content; // "This system software update improves system performance."
+	};
+
+	/** Represents the "package" json object */
+	struct pkg_package
+	{
+		std::string version; // 01.04
+		int size = 0;
+		std::string sha1sum;        // a5c83b88394ea3ae99974caedd38690981a80f3e
+		std::string ps3_system_ver; // 04.4000
+		std::string drm_type;       // local or mbind etc.
+		std::vector<pkg_changelog> changelogs;
+		std::vector<pkg_title> titles;
+
+		std::string get_changelog(const std::string& type) const
+		{
+			if (auto it = std::find_if(changelogs.begin(), changelogs.end(), [type](const pkg_changelog& cl) { return cl.type == type; });
+				it != changelogs.end())
+			{
+				return it->content;
+			}
+			if (auto it = std::find_if(changelogs.begin(), changelogs.end(), [](const pkg_changelog& cl) { return cl.type == "paramhip"; });
+				it != changelogs.end())
+			{
+				return it->content;
+			}
+			return "";
+		}
+
+		std::string get_title(const std::string& type) const
+		{
+			if (auto it = std::find_if(titles.begin(), titles.end(), [type](const pkg_title& t) { return t.type == type; });
```

Comment:
```suggestion
			if (auto it = std::find_if(titles.begin(), titles.end(), [&type](const pkg_title& t) { return t.type == type; });
```
Huge nitpick, as AFAIK capturing references with value semantics copies them, and we're fine with a reference here.